### PR TITLE
Orlp-419 WIP: Add Javadocs to all interfaces (first part)

### PR DESCRIPTION
**Pull request by VovaChornyi**

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/service/AccountService.java [hunk 1]
```diff
@@ -1,30 +1,93 @@
...
 import java.util.List;
 
+/**
+ * Works with accounts of users
+ */
 public interface AccountService {
+    /**
+     * Updates account
+     *
+     * @param account updated account
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/AccountService.java]: update

File: src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java [hunk 2]
```diff
@@ -11,29 +11,104 @@
...
+     * Returns cards that belong deck with the given identifier depends on learning regime.
+     *
+     * @param deckId must not be {@literal null}.
+     * @return all cards with the given deck's id (by default 10)
+     * @throws NotAuthorisedUserException
+     */
     List<Card> getLearningCards(Long deckId) throws NotAuthorisedUserException;
 
+    /**
+     * Retrieves an card by its id.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java]: by the given identifier

File: src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java [hunk 3]
```diff
@@ -11,29 +11,104 @@
...
+     *
+     * @param id must not be {@literal null}.
+     * @return the card with the given id or {@literal null} if none found.
+     */
     Card getCard(Long id);
 
+    /**
+     * Adds card to deck with the given identifier.
+     *
+     * @param card   must not be {null}.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java]: Maybe need to add more description for params.
We can discuss it.

File: src/main/java/com/softserve/academy/spaced/repetition/service/CourseCommentService.java [hunk 4]
```diff
@@ -6,15 +6,51 @@
...
+    /**
+     * Gets all comments of the course with the given identifier.
+     *
+     * @param courseId - course's id, must not be {@literal null}.
+     * @return list of comments
+     */
     List<Comment> getAllCommentsForCourse(Long courseId);
 
+    /**
+     * Updates comment with the given identifier on the new text
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/CourseCommentService.java]: Update text of comment with the given identifier ( maybe )

File: src/main/java/com/softserve/academy/spaced/repetition/service/CourseService.java [hunk 5]
```diff
@@ -7,39 +7,151 @@
...
+
+    /**
+     * Find all published courses.
+     *
+     * @return list of courses
+     */
     List<Course> getAllCourses();
 
+    /**
+     * Find all published courses in category with given identifier.
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/service/CourseService.java]: with the given identifier of category

File: src/main/java/com/softserve/academy/spaced/repetition/service/AccountService.java (hunk 1 above)
Comment (VovaChornyi) [reply src/main/java/com/softserve/academy/spaced/repetition/service/AccountService.java]: no
it use fot method context, so it must be updates

File: src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java (hunk 2 above)
Comment (VovaChornyi) [reply src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java]: ok

File: src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java (hunk 3 above)
Comment (VovaChornyi) [reply src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java]: ok

File: src/main/java/com/softserve/academy/spaced/repetition/service/CourseCommentService.java (hunk 4 above)
Comment (VovaChornyi) [reply src/main/java/com/softserve/academy/spaced/repetition/service/CourseCommentService.java]: ok your comment is much better

File: src/main/java/com/softserve/academy/spaced/repetition/service/CourseService.java (hunk 5 above)
Comment (VovaChornyi) [reply src/main/java/com/softserve/academy/spaced/repetition/service/CourseService.java]: my comment is also correct

File: src/main/java/com/softserve/academy/spaced/repetition/service/AccountService.java (hunk 1 above)
Comment (MoGUREV) [reply src/main/java/com/softserve/academy/spaced/repetition/service/AccountService.java]: Okay, write updates)

**Pull request merged**